great_expectations/core/batch_manager.py

```diff
@@ -0,0 +1,166 @@
+import logging
+from collections import OrderedDict
+from typing import Dict, List, Optional
+
+from great_expectations.core.batch import (
+    Batch,
+    BatchData,
+    BatchDefinition,
+    BatchMarkers,
+)
+from great_expectations.core.id_dict import BatchSpec
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+
+
+class BatchManager:
+    def __init__(
+        self,
+        execution_engine: "ExecutionEngine",  # noqa: F821
+        batch_list: Optional[List[Batch]] = None,
```

Comment:
Why default `batch_list` to `None` just to immediately reset it to `[]`?

Reply:
@NathanFarmer Because doing so in constructor arguments generates a security alert from `PyCharm`.

Reply:
@NathanFarmer Unfortunately, default function parameters are evaluated at class definition time. That means, setting a default argument to a mutable object or container type like `[]` or `{}` can lead to some strange behavior:
```
>>> def f(a=[]):
...         a.append(1)
...         print(a)
... 
>>> f()
[1]
>>> f()
[1, 1]
>>> f()
[1, 1, 1]
>>> f(a=[1,2,3])
[1, 2, 3, 1]
>>> f()
[1, 1, 1, 1]
```

Reply:
@billdirks that makes sense, thanks!

Reply:
This is actually a handy "feature" when doing recursive functions.